core/test/qa_SchedulerMessages.cpp

```diff
@@ -25,13 +25,23 @@ TestContext* context = new TestContext(paths{}, // plugin paths
     gr::blocklib::initGrTestingBlocks);
 
 class TestScheduler {
-public:
-    TestScheduler(gr::Graph graph) : scheduler_(std::move(graph)) {
-        using namespace gr::testing;
+    using TScheduler = gr::scheduler::Simple<gr::scheduler::ExecutionPolicy::singleThreaded>;
+    std::expected<void, gr::Error> schedulerRet_;
+    std::thread                    schedulerThread_;
 
-        scheduler_.graph().emplaceBlock<gr::testing::SlowSource<float>>();
-        scheduler_.graph().emplaceBlock<gr::testing::CountingSink<float>>();
+    gr::Graph& addSourceSink(gr::Graph& graph) const noexcept {
```

Comment:
`gr::Graph&& addSourceSink(gr::Graph&& graph) const noexcept {` (rvalue references)

Also, the name could have been nicer -- `withTestingSourceAndSink` or similar. `addSourceSink` sounds like it adds a source-sink :)